src/main/java/it/pagopa/pn/deliverypush/service/impl/WebhookStreamsServiceImpl.java

```diff
@@ -36,43 +39,56 @@ public class WebhookStreamsServiceImpl extends WebhookServiceImpl implements Web
+
+    private int maxStreams;
+    private int purgeDeletionWaittime;
+
+    public WebhookStreamsServiceImpl (StreamEntityDao streamEntityDao, SchedulerService schedulerService
+        , PnDeliveryPushConfigs pnDeliveryPushConfigs, PnExternalRegistryClient pnExternalRegistryClient){
         super(streamEntityDao);
         this.schedulerService = schedulerService;
         this.pnDeliveryPushConfigs = pnDeliveryPushConfigs;
+        this.pnExternalRegistryClient = pnExternalRegistryClient;
     }
 
-    private int maxStreams;
-    private int purgeDeletionWaittime;
-
     @PostConstruct
     private void postConstruct() {
         PnDeliveryPushConfigs.Webhook webhookConf = pnDeliveryPushConfigs.getWebhook();
-        this.maxStreams = webhookConf.getScheduleInterval().intValue();
+        this.maxStreams = webhookConf.getMaxStreams().intValue();
         this.purgeDeletionWaittime = webhookConf.getPurgeDeletionWaittime();
     }
     @Override
-    public Mono<StreamMetadataResponseV23> createEventStream(String xPagopaPnCxId, List<String> xPagopaPnCxGroups, String xPagopaPnApiVersion, Mono<StreamCreationRequestV23> streamCreationRequest) {
+    public Mono<StreamMetadataResponseV23> createEventStream(String xPagopaPnUid, String xPagopaPnCxId, List<String> xPagopaPnCxGroups, String xPagopaPnApiVersion, Mono<StreamCreationRequestV23> streamCreationRequest) {
+        String msg = "createEventStream xPagopaPnCxId={}, xPagopaPnCxGroups={}, xPagopaPnApiVersion={} ";
+        String[] args = new String[] {xPagopaPnCxId, groupString(xPagopaPnCxGroups), xPagopaPnApiVersion};
 
         return streamCreationRequest.doOnNext(payload-> {
-            String msg = "createEventStream xPagopaPnCxId={}, xPagopaPnCxGroups={}, xPagopaPnApiVersion={}, payload={}";
-            String[] args = new String[] {xPagopaPnCxId, groupString(xPagopaPnCxGroups), xPagopaPnApiVersion, payload.toString()};
             generateAuditLog(PnAuditLogEventType.AUD_WH_CREATE, msg, args).log();
-        }).map(r -> Tuples.of (DtoToEntityStreamMapper.dtoToEntity(xPagopaPnCxId, UUID.randomUUID().toString(), r)
+        }).flatMap(x->
+                (x.getReplacedStreamId() == null ? checkStreamCount(xPagopaPnCxId) : Mono.just(Boolean.TRUE)).then(Mono.just(x))
+        ).map(r -> Tuples.of (
+            DtoToEntityStreamMapper.dtoToEntity(xPagopaPnCxId, UUID.randomUUID().toString(), r)
                 , r.getReplacedStreamId() != null ? r.getReplacedStreamId().toString() : ""))
             .flatMap(t2 -> {
-                return WebhookUtils.checkGroups(t2.getT1().getGroups(),xPagopaPnCxGroups)?
-                    (StringUtils.isBlank(t2.getT2()) ? streamEntityDao.save(t2.getT1()) : streamEntityDao.replace(t2.getT1(), t2.getT2()))
+                List<String> allowedGroups = (xPagopaPnCxGroups==null || xPagopaPnCxGroups.isEmpty())
+                    ? pnExternalRegistryClient.getGroups(xPagopaPnUid, xPagopaPnCxId)
+                    : t2.getT1().getGroups();
```

Comment:
qui credo ci vada xpagopapncxgroups (siamo nel ramo in cui l'array non è vuoto e quindi le allowed saranno quelle dell'apikey stessa).
Inoltre, ti ricordo che per la v10, il check sui gruppi non avviene (nel DTO saran sempre vuoti, e un'apikey qualsiasi può creare stream v10)
Tanto poi quando implementerai i prossimi metodi, farai refactor per metterlo a fattor comune su una funzioncina dedicata.
E questa funzioncina, potremmo anche valutare un WebhookValidationUtils, in cui mettere tutti i check vari ed eventuali (il check sui max, sulle versioni, sui gruppi, ecc ecc

Reply:
se vedi controllo proprio quelle dello xpagopapncxgroups . Ok invece per v10 non mi era chiaro lo skip